components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/CustomTabIntentProcessor.kt

```diff
@@ -31,6 +33,20 @@ class CustomTabIntentProcessor(
         return safeIntent.action == ACTION_VIEW && isCustomTabIntent(safeIntent)
     }
 
+    @VisibleForTesting
+    internal fun getAdditionalHeaders(intent: SafeIntent): Map<String, String>? {
+        val pairs = intent.getBundleExtra(Browser.EXTRA_HEADERS)
+        val headers = mutableMapOf<String, String>()
+        pairs?.keySet()?.forEach { key ->
+            headers[key] = pairs.getString(key)!!
```

Comment:
Do we really want to throw here on null or maybe just ignore null values (or log)?